fullstop-jobs/src/main/java/org/zalando/stups/fullstop/jobs/ami/FetchAmiJob.java

```diff
@@ -0,0 +1,203 @@
+
+                    for (final Reservation reservation : describeEC2Result.getReservations()) {
+
+                        for (final Instance instance : reservation.getInstances()) {
+                            final Map<String, Object> metaData = newHashMap();
+                            final List<String> errorMessages = newArrayList();
+
+                            if (violationService.violationExists(account, region, EVENT_ID, instance.getInstanceId(), OUTDATED_AMI)) {
+                                continue;
+                            }
+
+                            Optional<Image> image = getAmiFromEC2Api(account, region, instance.getImageId()).get().stream().findFirst();
+                            Optional<Boolean> isTaupageAmi = image
+                                    .filter(img -> img.getName().startsWith(taupageNamePrefix))
+                                    .map(Image::getOwnerId)
+                                    .map(taupageOwners::contains);
+
+                            // will not check for all non taupage ami
+                            // or images with taupage as name but created from another owner
+                            if (!isTaupageAmi.orElse(false)) {
+                                continue;
+                            }
+
+                            DateTime now = DateTime.now();
+                            if (isTaupageTooOld(image.map(Image::getName), now)) {
+                                metaData.put("ami_owner_id", image.map(Image::getOwnerId).orElse(""));
+                                metaData.put("ami_id", image.map(Image::getImageId).orElse(""));
+                                metaData.put("ami_name", image.map(Image::getName).orElse(""));
+                                DateTime maxValidityTimeForAmi = now.minus(Days.days(60));
+                                errorMessages.add("Outdated AMI! Please use the latest version available." +
+                                        " Should be at least from: " + maxValidityTimeForAmi.monthOfYear().getAsText()
+                                        + " " + maxValidityTimeForAmi.year().getAsText());
+                            }
+
+                            if (metaData.size() > 0) {
+                                metaData.put("errorMessages", errorMessages);
+                                writeViolation(account, region, metaData, instance.getInstanceId());
+                            }
+
+                        }
+
+                    }
+
+                } catch (AmazonServiceException a) {
+
+                    if (a.getErrorCode().equals("RequestLimitExceeded")) {
+                        log.warn("RequestLimitExceeded for account: {}", account);
+                    } else {
+                        log.error(a.getMessage(), a);
+                    }
+
+                }
+            }
+        }
+    }
+
+    private boolean isTaupageTooOld(Optional<String> imageName, DateTime now) {
```

Comment:
as I mentioned above, I would refactor this method to accept an `Image` and return the expirationDate of this image.